version/version.go

```diff
@@ -34,28 +34,34 @@ var (
 	Commit = ""
 	// build tags
 	BuildTags = ""
+	// cosmos-sdk version
+	CosmosSDKVersion = defaultCosmosSDKVersion
 )
 
+const defaultCosmosSDKVersion = "0.42.5"
+
```

Comment:
default should be empty IMHO.

```suggestion
	CosmosSDKVersion = ""
)

```

Reply:
This is to avoid using ldflags which almost no one sets. The default is bumped by the release manager, but it can be overridden by an ldflag if need be

Reply:
Personally, I don't think we should be doing that. We should encourage that flag being set. It's more dangerous to forget and update this value rather than forgetting to set an LDFlag.

If this value is empty, we can print an warning log.

Reply:
I am mixed on the approach. I agree with you, it should be on the app developer, that is the approach I moved tendermiint to. Then people started complaining about it, and most people skip that changelog. Adding an extra step to the release process is easier than having to keep answering questions. 

I can leave this open for a few days to see what others think.

Reply:
This should be automated. I agree with @alexanderbez  - it's error prone to remember to update this value.
if we use `git describe` here, then there is no extra step and it will work for all branches.

Reply:
@robert-zaremba how do you get the version of something that hasnt been tagged yet?

Reply:
There are two "bad" situations that can happen -- you have to pick one:

1. The app developer accidentally forgets to set `CosmosSDKVersion` and when deployed on a network, could reflect a completely inaccurate version causing UX confusion and needless debugging.
2. The app developer accidentally forgets to set LDFLAGS and thus the version is empty.

IMHO (1) is much worse than (2), and so why that's why I think we should NOT have a default. Make sure LDFLAGS are set, if they're not, we can simply print a warning during the build/compile process.

Reply:
Would https://golang.org/pkg/runtime/debug/#ReadBuildInfo work as an automated solution?

Reply:
does this work if the dependency is looking for modules or only if the top level app?

Reply:
bumping this question: any active qualms still against this PR?

Reply:
I can play around with the build info, but not sure if it will work, as I don't know if a dep and read the apps go.mod.

Reply:
@marbar3778 how's this going?

Reply:
on vacation this week. Back next week. This is at the top of my list.

Reply:
@marbar3778 I hope you enjoy your vacation!

Reply:
went with buildinfo and grabbing it. @alexanderbez @robert-zaremba @amaurym can you review again